src/test/java/bio/overture/ego/token/JwtTest.java

```diff
@@ -79,4 +92,105 @@ public void jwtContainsUserGroups() {
     assertEquals(1, groups.size());
     assertTrue(groups.contains(groupId));
   }
+
+  @Test
+  @SneakyThrows
+  public void applicationPermsOnly_appJwtHasAllResolvedScopes_Success() {
+    val app = entityGenerator.setupApplication("TestApp", "testsecret", ApplicationType.CLIENT);
+    val policies = entityGenerator.setupPolicies("SONG", "SCORE", "DACO");
+
+    val permReq1 = PermissionRequest.builder().policyId(policies.get(0).getId()).mask(READ).build();
+    val permReq2 =
+        PermissionRequest.builder().policyId(policies.get(1).getId()).mask(WRITE).build();
+    val permReq3 = PermissionRequest.builder().policyId(policies.get(2).getId()).mask(DENY).build();
+
+    // add app permissions
+    applicationPermissionService.addPermissions(
+        app.getId(), Arrays.asList(permReq1, permReq2, permReq3));
+    val appClient = applicationService.loadClientByClientId(app.getClientId());
+
+    // get permissions, assert count matches
+    val appPerms = applicationService.getByClientId(app.getClientId()).getApplicationPermissions();
+    assertEquals(appPerms.size(), 3);
+
+    val permScopes =
+        appPerms.stream()
+            .map(AbstractPermissionService::buildScope)
+            .map(Scope::toString)
+            .collect(Collectors.toList());
```

Comment:
since the usage of `permScopes` doesnt require order, use
`.collect(toSet());`

instead